src/lib.rs

```diff
@@ -290,14 +292,29 @@ pub struct Env {
 /// A context passed to paint methods of widgets.
 ///
 /// Widgets paint their appearance by calling methods on the
-/// `render_ctx`. This struct is expected to grow, for example to
-/// include the "damage region" indicating that only a subset of
-/// the entire widget hierarchy needs repainting.
+/// `render_ctx`, which PaintCtx derefs to for convenience.
+/// This struct is expected to grow, for example to include the
+/// "damage region" indicating that only a subset of the entire
+/// widget hierarchy needs repainting.
 pub struct PaintCtx<'a, 'b: 'a> {
     /// The render context for actually painting.
     pub render_ctx: &'a mut Piet<'b>,
 }
 
+impl<'a, 'b: 'a> Deref for PaintCtx<'a, 'b> {
+    type Target = &'a mut Piet<'b>;
```

Comment:
Actually I had a closer look, and it should just be `type Target = Piet<'b>`. Then the body needs no additional deref, it's just `&self.render_ctx`.

This is very subtle because it does work as written, but not as efficiently. What you get with the current state of the code is a reference to the field within the `PaintCtx` object. You can still call methods on it because magic deref will follow both references, but it's a double indirection (although probably will get optimized).

Reply:
Hmm, this is subtle, I wouldn't have thought of it. Although having the target of `Deref` (not `DerefMut`) be `&mut` maybe looks a little strange.

Reply:
Thank you for this explanation! Yeah, I should've taken more pause at the &mut in the regular Deref.